src/docs/asciidoc/installing_upgrading.adoc

```diff
@@ -0,0 +1,771 @@
+[[installing-hazelcast-imdg]]
+== Installing and Upgrading
+
+This chapter provides pre- and post-installation
+details, and deployment options of Hazelcast IMDG.
+
+=== Using Maven
+
+You can find Hazelcast in standard Maven repositories. If your
+project uses Maven, you do not need to add
+additional repositories to your `pom.xml` or add
+`hazelcast-{imdg-version}.jar` file into your
+classpath (Maven does that for you). Just add the following
+lines to your `pom.xml`:
+
+
+[source,xml,indent=0,subs="verbatim,attributes+",role="primary"]
+.Hazelcast Open Source Edition
+----
+<dependencies>
+    <dependency>
+        <groupId>com.hazelcast</groupId>
+        <artifactId>hazelcast</artifactId>
+        <version>{imdg-version}</version>
+    </dependency>
+</dependencies>
+----
+
+[source,xml,indent=0,subs="verbatim,attributes+",role="secondary"]
+.Hazelcast Enterprise Edition
+----
+<!-- There are two Maven repositories defined for Hazelcast IMDG Enterprise: -->
+
+<repository>
+    <id>Hazelcast Private Snapshot Repository</id>
+    <url>https://repository.hazelcast.com/snapshot/</url>
+</repository>
+<repository>
+    <id>Hazelcast Private Release Repository</id>
+    <url>https://repository.hazelcast.com/release/</url>
+</repository>
+
+
+<!-- You may also define dependencies: -->
+
+<dependency>
+    <groupId>com.hazelcast</groupId>
+    <artifactId>hazelcast-enterprise</artifactId>
+    <version>{imdg-version}</version>
+</dependency>
+<dependency>
+    <groupId>com.hazelcast</groupId>
+    <artifactId>hazelcast-enterprise-all</artifactId>
+    <version>{imdg-version}</version>
+</dependency>
+<dependency>
+    <groupId>com.hazelcast</groupId>
+    <artifactId>hazelcast-enterprise-all</artifactId>
+    <version>{imdg-version}</version>
+    <classifier>javadoc</classifier>
+</dependency>
+----
+
+Above dependency (`hazelcast`) includes both member and Java
+client libraries of Hazelcast IMDG. A separate Java client module
+and dependency do not exist. See <<removal-of-hazelcast-client-module, here>>
+for the details.
+
+=== Using the Download Archives
+
+As an alternative to Maven, you can download and install Hazelcast IMDG
+yourself. You only need to:
+
+* download the package `hazelcast-{imdg-version}.zip` or `hazelcast-{imdg-version}.tar.gz`
+from link:https://hazelcast.org/download[hazelcast.org^]
+* extract the downloaded `hazelcast-{imdg-version}.zip` or `hazelcast-{imdg-version}.tar.gz`
+* and add the file `hazelcast-{imdg-version}.jar` to your classpath.
+
+[[using-the-scripts-in-the-package]]
+==== Using the Scripts In The Package
```

Comment:
This was a bit strange, seamlessly transitioning between downloading to starting at the same level. WDYT?